OpenNameSearch in Ubuntu 16.04 fresh install fail

Scanning dependencies of target osm2pgsql_lib
[  2%] Building CXX object osm2pgsql/CMakeFiles/osm2pgsql_lib.dir/expire-tiles.cpp.o
[  4%] Building CXX object osm2pgsql/CMakeFiles/osm2pgsql_lib.dir/geometry-processor.cpp.o
[  7%] Building CXX object osm2pgsql/CMakeFiles/osm2pgsql_lib.dir/id-tracker.cpp.o
[  9%] Building CXX object osm2pgsql/CMakeFiles/osm2pgsql_lib.dir/middle-pgsql.cpp.o
[ 11%] Building CXX object osm2pgsql/CMakeFiles/osm2pgsql_lib.dir/middle-ram.cpp.o
[ 14%] Building CXX object osm2pgsql/CMakeFiles/osm2pgsql_lib.dir/middle.cpp.o
[ 16%] Building CXX object osm2pgsql/CMakeFiles/osm2pgsql_lib.dir/node-persistent-cache.cpp.o
[ 19%] Building CXX object osm2pgsql/CMakeFiles/osm2pgsql_lib.dir/node-ram-cache.cpp.o
[ 21%] Building CXX object osm2pgsql/CMakeFiles/osm2pgsql_lib.dir/options.cpp.o
[ 23%] Building CXX object osm2pgsql/CMakeFiles/osm2pgsql_lib.dir/osmdata.cpp.o
[ 26%] Building CXX object osm2pgsql/CMakeFiles/osm2pgsql_lib.dir/osmium-builder.cpp.o
[ 28%] Building CXX object osm2pgsql/CMakeFiles/osm2pgsql_lib.dir/output-gazetteer.cpp.o
[ 30%] Building CXX object osm2pgsql/CMakeFiles/osm2pgsql_lib.dir/output-multi.cpp.o
[ 33%] Building CXX object osm2pgsql/CMakeFiles/osm2pgsql_lib.dir/output-null.cpp.o
[ 35%] Building CXX object osm2pgsql/CMakeFiles/osm2pgsql_lib.dir/output-pgsql.cpp.o
[ 38%] Building CXX object osm2pgsql/CMakeFiles/osm2pgsql_lib.dir/output.cpp.o
[ 40%] Building CXX object osm2pgsql/CMakeFiles/osm2pgsql_lib.dir/parse-osmium.cpp.o
[ 42%] Building CXX object osm2pgsql/CMakeFiles/osm2pgsql_lib.dir/pgsql.cpp.o
[ 45%] Building CXX object osm2pgsql/CMakeFiles/osm2pgsql_lib.dir/processor-line.cpp.o
[ 47%] Building CXX object osm2pgsql/CMakeFiles/osm2pgsql_lib.dir/processor-point.cpp.o
[ 50%] Building CXX object osm2pgsql/CMakeFiles/osm2pgsql_lib.dir/processor-polygon.cpp.o
[ 52%] Building CXX object osm2pgsql/CMakeFiles/osm2pgsql_lib.dir/reprojection.cpp.o
[ 54%] Building CXX object osm2pgsql/CMakeFiles/osm2pgsql_lib.dir/sprompt.cpp.o
[ 57%] Building CXX object osm2pgsql/CMakeFiles/osm2pgsql_lib.dir/table.cpp.o
[ 59%] Building CXX object osm2pgsql/CMakeFiles/osm2pgsql_lib.dir/taginfo.cpp.o
[ 61%] Building CXX object osm2pgsql/CMakeFiles/osm2pgsql_lib.dir/tagtransform.cpp.o
[ 64%] Building CXX object osm2pgsql/CMakeFiles/osm2pgsql_lib.dir/tagtransform-c.cpp.o
[ 66%] Building CXX object osm2pgsql/CMakeFiles/osm2pgsql_lib.dir/util.cpp.o
[ 69%] Building CXX object osm2pgsql/CMakeFiles/osm2pgsql_lib.dir/wildcmp.cpp.o
[ 71%] Linking CXX static library libosm2pgsql.a
[ 71%] Built target osm2pgsql_lib
Scanning dependencies of target osm2pgsql
[ 73%] Building CXX object osm2pgsql/CMakeFiles/osm2pgsql.dir/osm2pgsql.cpp.o
[ 76%] Linking CXX executable osm2pgsql
[ 76%] Built target osm2pgsql
Scanning dependencies of target nominatim_lib
[ 78%] Running external makefile /usr/lib/postgresql/9.5/lib/pgxs/src/makefiles/pgxs.mk
/home/ntim/Nominatim/module/Makefile:11: warning: overriding recipe for target 'install'
/usr/lib/postgresql/9.5/lib/pgxs/src/makefiles/pgxs.mk:123: warning: ignoring old recipe for target 'install'
[ 78%] Built target nominatim_lib
Scanning dependencies of target nominatim
[ 80%] Building C object nominatim/CMakeFiles/nominatim.dir/export.c.o
[ 83%] Building CXX object nominatim/CMakeFiles/nominatim.dir/geometry.cpp.o
[ 85%] Building C object nominatim/CMakeFiles/nominatim.dir/import.c.o
[ 88%] Building C object nominatim/CMakeFiles/nominatim.dir/index.c.o
[ 90%] Building C object nominatim/CMakeFiles/nominatim.dir/input.c.o
[ 92%] Building C object nominatim/CMakeFiles/nominatim.dir/nominatim.c.o
[ 95%] Building C object nominatim/CMakeFiles/nominatim.dir/postgresql.c.o
[ 97%] Building C object nominatim/CMakeFiles/nominatim.dir/sprompt.c.o
[100%] Linking CXX executable nominatim
[100%] Built target nominatim
ALTER ROLE
--2018-05-25 02:48:47--  http://download.geofabrik.de/europe/switzerland-latest.osm.pbf
Resolving download.geofabrik.de (download.geofabrik.de)... 88.99.24.113
Connecting to download.geofabrik.de (download.geofabrik.de)|88.99.24.113|:80... connected.
HTTP request sent, awaiting response... 200 OK
Length: 293473161 (280M) [application/octet-stream]
Saving to: ‘switzerland-latest.osm.pbf’

switzerland-latest.osm.pbf        100%[===========================================================>] 279.88M  72.8MB/s    in 4.0s

2018-05-25 02:48:51 (70.8 MB/s) - ‘switzerland-latest.osm.pbf’ saved [293473161/293473161]

--2018-05-25 02:48:51--  http://www.nominatim.org/data/country_grid.sql.gz
Resolving www.nominatim.org (www.nominatim.org)... 81.169.170.39
Connecting to www.nominatim.org (www.nominatim.org)|81.169.170.39|:80... connected.
HTTP request sent, awaiting response... 200 OK
Length: 42146983 (40M) [application/x-gzip]
Saving to: ‘/home/ntim/Nominatim/data/country_osm_grid.sql.gz’

/home/ntim/Nominatim/data/country 100%[===========================================================>]  40.19M  11.2MB/s    in 3.7s

2018-05-25 02:48:55 (11.0 MB/s) - ‘/home/ntim/Nominatim/data/country_osm_grid.sql.gz’ saved [42146983/42146983]

2018-05-25 02:48:55 == Create DB
2018-05-25 02:48:56 == Setup DB
Postgres version found: 9.5
Postgis version found: 2.2
2018-05-25 02:49:00 == WARNING: external UK postcode table not found.
2018-05-25 02:49:00 == Import data
osm2pgsql version 0.95.0-dev (64 bit id space)

Using projection SRS 4326 (Latlong)
NOTICE:  table "place" does not exist, skipping
Allocating memory for dense node cache
Allocating dense node cache in one big chunk
Out of memory for dense node cache, reduce --cache size
Error occurred, cleaning up
ERROR: Error executing external command: /home/ntim/Nominatim/build/osm2pgsql/osm2pgsql -lsc -O gazetteer --hstore --number-processes                                 1 -C 12106 -P 5432 -d nominatim /home/ntim/switzerland-latest.osm.pbf
Error executing external command: /home/ntim/Nominatim/build/osm2pgsql/osm2pgsql -lsc -O gazetteer --hstore --number-processes 1 -C 12                                106 -P 5432 -d nominatim /home/ntim/switzerland-latest.osm.pbf
Enabling conf nominatim_dir.
To activate the new configuration, you need to run:
  service apache2 reload
GRANT
GRANT
Collecting osmium
  Downloading https://files.pythonhosted.org/packages/7a/eb/63d7af416ee6126c96a3e5c057750e266da5d1eecc108a6fe87cad9e98c9/osmium-2.14.1                                .tar.gz (1.6MB)
    100% |¦¦¦¦¦¦¦¦¦¦¦¦¦¦¦¦¦¦¦¦¦¦¦¦¦¦¦¦¦¦¦¦| 1.6MB 665kB/s
    Complete output from command python setup.py egg_info:
    Traceback (most recent call last):
      File "<string>", line 1, in <module>
    ImportError: No module named setuptools

    ----------------------------------------
Command "python setup.py egg_info" failed with error code 1 in /tmp/pip-build-W9iVlY/osmium/
You are using pip version 8.1.1, however version 10.0.1 is available.
You should consider upgrading via the 'pip install --upgrade pip' command.
2018-05-25 02:49:08 == Index ranks 0 - 4
nominatim version 3.1.0

Failed preparing index_sectors: ERROR:  relation "placex" does not exist
LINE 1: select geometry_sector,count(*) from placex where rank_searc...
                                             ^

ERROR: Error executing external command: /home/ntim/Nominatim/build/nominatim/nominatim -i -d nominatim -P 5432 -t 1 -R 4
Error executing external command: /home/ntim/Nominatim/build/nominatim/nominatim -i -d nominatim -P 5432 -t 1 -R 4
2018-05-25 02:49:08 == Create Functions
NOTICE:  type "addressline" does not exist, skipping
Summary of warnings:


2018-05-25 02:49:08 == Setup finished.
2018-05-25 02:49:08 == Create Functions
NOTICE:  drop cascades to function get_addressdata(bigint,integer)
NOTICE:  drop cascades to function get_wikipedia_match(hstore,character varying)
Summary of warnings:


2018-05-25 02:49:08 == Setup finished.
PHP Warning:  file_get_contents(https://www.openstreetmap.org/api/0.6/node/DB Error: no such table/1): failed to open stream: HTTP req                                uest failed! HTTP/1.1 400 Bad Request
 in /home/ntim/Nominatim/lib/lib.php on line 42
ERROR: Cannot determine date of database.
Cannot determine date of database.
Created symlink from /etc/systemd/system/multi-user.target.wants/nominatum-updates.service to /etc/systemd/system/nominatum-updates.se                                rvice.

--- a/OpenNameSearch-Ubuntu16.sh
+++ b/OpenNameSearch-Ubuntu16.sh
@@ -171,7 +171,8 @@ function import_osm_data(){
     fi
 
     NP=$(grep -c 'model name' /proc/cpuinfo)
-    let C_MEM=$(free -m | grep -i 'mem:' | sed 's/[ \t]\+/ /g' | cut -f4,7 -d' ' | tr ' ' '+')-200
+    let AVAIL_MEM=$(free -m | grep -i 'mem:' | sed 's/[ \t]\+/ /g' | cut -f4,7 -d' ' | tr ' ' '+')
+    let C_MEM=(AVAIL_MEM/4)*3
     su - ${NM_USER} <<EOF
 cd /home/${NM_USER}/Nominatim/build
 wget -O /home/${NM_USER}/Nominatim/data/country_osm_grid.sql.gz http://www.nominatim.org/data/country_grid.sql.gz

--- a/OpenNameSearch.sh
+++ b/OpenNameSearch.sh
@@ -179,7 +179,8 @@ function import_osm_data(){
 	fi
  
 	NP=$(grep -c 'model name' /proc/cpuinfo)
-	let C_MEM=$(free -m | grep -i 'mem:' | sed 's/[ \t]\+/ /g' | cut -f4,7 -d' ' | tr ' ' '+')-200
+	let AVAIL_MEM=$(free -m | grep -i 'mem:' | sed 's/[ \t]\+/ /g' | cut -f4,7 -d' ' | tr ' ' '+')
+	let C_MEM=(AVAIL_MEM/4)*3
 	su - ${NM_USER} <<EOF
 cd /home/${NM_USER}/Nominatim-${NM_VER}
 ./utils/setup.php --osm-file ${PBF_FILE} --all --osm2pgsql-cache ${C_MEM} 2>&1 | tee setup.log

